addons/website_forum/models/forum_forum.py

```diff
@@ -130,6 +130,20 @@ def _get_default_welcome_message(self):
     karma_user_bio = fields.Integer(string='Display detailed user biography', default=750)
     karma_post = fields.Integer(string='Ask questions without validation', default=100)
     karma_moderate = fields.Integer(string='Moderate posts', default=1000)
+    has_pending_post = fields.Boolean(string='Has pending post', compute="_compute_has_pending_post")
+
+    @api.depends_context('uid')
+    def _compute_has_pending_post(self):
+        domain = [
+            ('create_uid', '=', self.env.user.id),
+            ('state', '=', 'pending'),
+            ('parent_id', '=', False),
+            ('forum_id', 'in', self.ids)
+        ]
+        data = self.env['forum.post'].read_group(domain, ['forum_id'], ['forum_id'])
+        result = {item['forum_id'][0]: item['forum_id_count'] for item in data}
```

Comment:
@flch-odoo  i will let the code be this way. thanks

Reply:
indeed I prefer the `_read_group` version rather than `read_group`

but even better, you don't even need to do a `GROUP BY`, a single `SELECT` is enough
```py
@api.depends_context('uid')
def _compute_has_pending_post(self):
    domain = [
        ('create_uid', '=', self.env.user.id),
        ('state', '=', 'pending'),
        ('parent_id', '=', False),
    ]
    result = self.env['forum.forum'].search([
        ('id', 'in', self.ids),
        ('post_ids', 'any', domain),
    ])
    result.has_pending_post = True
    (self - result).has_pending_post = False
```